code/modules/food_and_drinks/kitchen_machinery/griddle.dm

```diff
@@ -63,6 +63,37 @@
 
 
 /obj/machinery/griddle/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/storage/bag/tray))
+		var/datum/component/storage/tray_storage = I.GetComponent(/datum/component/storage)
+		if(!I.contents.len && griddled_objects.len > 0) //if the griddle has things on it and the tray is empty, put what you can into the tray
+			for(var/obj/item/grilled as anything in griddled_objects)
+				if(!griddled_objects.len)
+					return
+				if(I.contents.len >= tray_storage.max_items)
+					to_chat(user, span_notice("[I] can't fit more items!")) //tray is full
+					return
+				if(SEND_SIGNAL(I, COMSIG_TRY_STORAGE_INSERT, grilled, user))
+					ItemRemovedFromGrill(grilled)
+				else if(!SEND_SIGNAL(I, COMSIG_TRY_STORAGE_INSERT, grilled, user))
+					to_chat(user, span_notice("[I] can't hold [grilled]!")) //if an item cant enter the tray, dont touch it
+				else
+					return
+		else if(!I.contents.len)
```

Comment:
nah this is a different check entirely, the above check passes only if the griddle has items on it and the tray is empty, THIS check is actually for if BOTH the griddle and the tray are empty, since in that instance NEITHER of the other two ifs will actually pass, this if is to cover that situation (hence the chat message is "you cant griddle trays" because that if will only call if your attempting to place a tray onto an empty griddle)